tests/api/test_connection.py

```diff
@@ -254,6 +254,86 @@ def test_get_job_result(self, connection, result_dtype, monkeypatch):
 
         assert np.array_equal(result.samples, result_samples)
 
+    @pytest.mark.parametrize(
+        "result_dtype",
+        [
+            np.int8,
+            np.int16,
+            np.int32,
+            np.int64,
+            np.uint8,
+            np.uint16,
+            np.uint32,
+            np.uint64,
+            np.float32,
+            np.float64,
+        ],
+    )
```

Comment:
Should we also test complex number types?

Same goes for `test_get_job_result_dict_single_samples_array()` below.